### PR TITLE
[Paywalls V2] Update color spec

### PR DESCRIPTION
**Pull request by joshdholtz**
## Motivation

Updating color property to hold more types info:
1) Hex color
2) Alias to a project color

Note: Developers will soon™️  be able to define global colors for the project and paywall components will be able to reference them by name (alias)

ℹ️ This is A LOT of *struct*-ural changes to make this complie again but no functional changes were made

### Previously

Colors were strictly a hex color value

```json
{
    "light": "#ffffff",
    "dark": "#000000"
}
```

### Now

Colors can be a type of `hex` or `alias` (a name to a dictionary that will be coming down in the API response with all paywalls that is common on a project)

```json
{
    "light": {
        "type": "hex",
        "value": "#ffffff"
    },
    "dark": {
        "type": "hex",
        "value": "#000000"
    }
}
```

or

```json
{
    "light": {
        "type": "alias",
        "value": "primary"
    },
    "dark": {
        "type": "alias",
        "value": "primary_dark"
    }
}
```

### CODE REVIEW

File: Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift [hunk 1]
```diff
@@ -43,18 +43,80 @@ public extension PaywallComponent {
...
+        private enum ColorInfoTypes: String, Decodable {
+
+            case hex
+            case alias
+
+        }
+
+    }
+
+//    struct ColorInfo: Codable, Sendable, Hashable, Equatable {
```
Comment (fire-at-will) [inline Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift:108]: Nit: remove commented code

File: Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift [hunk 2]
```diff
@@ -43,18 +43,80 @@ public extension PaywallComponent {
...
+            case .hex(let hex):
+                try container.encode(ColorInfoTypes.hex.rawValue, forKey: .type)
+                try container.encode(hex, forKey: .value)
+            case .alias(let alias):
+                try container.encode(ColorInfoTypes.alias.rawValue, forKey: .type)
+                try container.encode(alias, forKey: .value)
+            }
+        }
+
+        public init(from decoder: Decoder) throws {
```
Comment (fire-at-will) [inline Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift:76]: I might think about adding in some tests to validate the custom encoding/decoding of the ColorInfo enum. I know from recent experience in Pines that it's easy to make mistakes in these custom enum codable implementations and it's easy to miss until much later in testing

**Review by fire-at-will — APPROVED** [review]
Looks good to me!

One question: Did you mean to remove `Tests/TestingApps/PaywallsTester/PaywallsTester/Config/SamplePaywalls.swift` and `Tests/TestingApps/PaywallsTester/PaywallsTester/UI/Views/SamplePaywallsList.swift`? I didn't see any mention of it in the PR description so I figured I'd double check

File: Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift (hunk 1 above)
Comment (joshdholtz) [reply Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift:108]: Ha! I did notice this and it gets removed like 2 PRs in 😛

File: Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift (hunk 2 above)
Comment (joshdholtz) [reply Sources/Paywalls/Components/Common/PaywallComponentPropertyTypes.swift:76]: Yes yes, for sure! Tests coming soon™️

**Pull request merged**